src/block-management/block-manager.js

```diff
@@ -260,6 +269,23 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 		return index === this.state.blocks.length - 1;
 	}
 
+	isEmptyBlock( block: BlockType ) {
+		const content = block.attributes.content;
+		const innerBlocks = block.innerBlocks;
+		return ( content === undefined || content === '' ) && ( innerBlocks.length === 0 );
+	}
+
+	isCandidateForReplaceBlock( block: BlockType ) {
+		return ( block.name === 'core/paragraph' || block.name === 'core/heading' || block.name === 'core/code' );
+	}
+
+	isReplaceable( block: ?BlockType ) {
+		if ( ! block ) {
+			return false;
+		}
+		return this.isEmptyBlock( block ) && this.isCandidateForReplaceBlock( block );
```

Comment:
Late to the party, but we should reuse the same logic that Gutenberg for this: `isUnmodifiedDefaultBlock` from `@wordpress/blocks`: https://github.com/WordPress/gutenberg/blob/40d1282a992702e1933118ef626f0666692baf2b/packages/editor/src/components/inserter/menu.js#L389

Reply:
Good catch! I miss that 👍

Reply:
Good catch indeed! opening a PR for this valuable change